WordPress/Classes/Services/CommentService+Likes.swift

```diff
@@ -40,46 +40,17 @@ extension CommentService {
                                                             commentID: commentID,
                                                             siteID: siteID,
                                                             purgeExisting: purgeExisting) {
-                                            let users = self.likeUsersFor(commentID: commentID, siteID: siteID)
+                                            assert(Thread.isMainThread)
+
+                                            let users = LikeUserHelper.likeUsersFor(commentID: commentID, siteID: siteID, in: self.coreDataStack.mainContext)
                                             success(users, totalLikes.intValue, count)
-                                            LikeUserHelper.purgeStaleLikes()
```

Comment:
This line is moved into the private `createNewUsers` function below. No other functions calls the `createNewUsers` function. So the only change here is swapping order of purging stale likes and calling the success closure. I don't think the order matters that much.